### DISCUSSION TITLE
Help Request: how can i use this lib for a 2D game?

### DISCUSSION THREAD

**Discussion by pheonixfirewingz** [jrouwe/JoltPhysics#1416] [Q&A]
I'm working on a 2D game that requires heavy physical hot can I use Jolt to achieve it

**Comment by jrouwe** [comment]
That's a very broad question. I would start by looking at the samples:

https://github.com/jrouwe/JoltPhysics/blob/master/Samples/Tests/General/TwoDFunnelTest.cpp

**Reply by pheonixfirewingz** [reply]
It’s a pure 2D game that doesn’t have a “Z” axis I wanted to find out how to use jolt when I don’t have a Z axis in play

**Reply by jrouwe** [reply]
You would just specify 0 for the Z axis. Jolt's vectors are always 3D, even if you're not using the Z coordinate. The line `bcs.mAllowedDOFs = EAllowedDOFs::Plane2D;` ensures that all bodies stay at Z = 0.

**Reply by pheonixfirewingz** [reply]
Hay I’m stuck at a point where i need to sync the jolt physics to the scripting & rendering  engine how can i get and set the position and rotate  pre & post jolt updates. I couldn’t find where there was an example of this.

**Reply by jrouwe** [reply]
Use `BodyInterface::GetPositionAndRotation` on e.g. `PhysicsSystem::GetActiveBodies`
